components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListener.java

```diff
@@ -382,4 +485,191 @@ private void removeOauthConsumerAppProperties(ServiceProvider serviceProvider, S
+                                break;
+                            }
+                        }
+                    } catch (IdentityOAuthAdminException e) {
+                        // Do nothing, the key does exists.
+                    }
+
+                    if ((oAuthAppDO.getGrantTypes().contains(OAuthConstants.GrantTypes.AUTHORIZATION_CODE)
+                            || oAuthAppDO.getGrantTypes().contains(OAuthConstants.GrantTypes.IMPLICIT))
+                            && StringUtils.isEmpty(oAuthAppDO.getCallbackUrl())) {
+                        validationMsg.add("Callback Url is required for Code or Implicit grant types");
+                    }
+
+                    validateScopeValidators(oAuthAppDO.getScopeValidators(), validationMsg);
+
+                    if (OAuthConstants.OAuthVersions.VERSION_2.equals(oAuthAppDO.getOauthVersion())) {
+                        validateGrants(oAuthAppDO.getGrantTypes().split("\\s"), validationMsg);
+                    }
+
+                    break;
+                }
+            }
+        }
+        if (!validationMsg.isEmpty()) {
+            throw new IdentityApplicationManagementValidationException(validationMsg.toArray(new String[0]));
+        }
+    }
+
+    /**
+     * Validate requested grants in the oauth app.
+     *
+     * @param requestedGrants list of requested grants
+     * @param validationMsg      validation msg list
+     */
+    private void validateGrants(String[] requestedGrants, List<String> validationMsg) {
+
+        OAuthAdminService oAuthAdminService = new OAuthAdminService();
+        List<String> allowedGrants = new ArrayList<>(Arrays.asList(oAuthAdminService.getAllowedGrantTypes()));
+        for (String requestedGrant : requestedGrants) {
+            if (StringUtils.isBlank(requestedGrant)) {
+                continue;
+            }
+            if (!allowedGrants.contains(requestedGrant)) {
+                validationMsg.add(String.format("Grant type %s not allowed", requestedGrant));
+            }
+        }
+    }
+
+    /**
+     * Validate scope validators in the oauth app.
+     *
+     * @param appScopeValidators list of scope validators
+     * @param validationMsg      validation msg list
+     */
+    private void validateScopeValidators(String[] appScopeValidators, List<String> validationMsg) {
+
+        OAuthAdminService oAuthAdminService = new OAuthAdminService();
+        List<String> scopeValidators = new ArrayList<>(Arrays.asList(oAuthAdminService.getAllowedScopeValidators()));
+        Arrays.stream(appScopeValidators).forEach(validator -> {
+            if (!scopeValidators.contains(validator)) {
+                validationMsg.add(String.format("The scope validator %s is not available in the " +
+                        "server configuration. ", validator));
+            }
+        });
+    }
+
+    /**
+     * Creates authenticated user obj form user obj.
```

Comment:
Correct the spelling mistake (**form**)

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/942/commits/ca4690d734709b108ef2df33a69ae3c9242febc8